src/materials/ConcreteExpansionEigenstrainBase.C

```diff
@@ -172,10 +174,9 @@ ConcreteExpansionEigenstrainBase::findNeighborIndex(Real sig)
     return 2;
   else if (sig > -_sigma_u && sig <= 0)
     return 1;
-  else if (sig > 0)
-    return 0;
   else
-    mooseError("Invalid value for sig.");
+    return 0;
+  mooseAssert("sig not real", "sig not real");
```

Comment:
Just delete line 179. The if, else if, and else cover all possible cases.

Reply:
Also, this function should return an unsigned int rather than an int.

Reply:
I will address this.